src/Cli/Command/Helper/DistinguishedNameHelper.php

```diff
@@ -33,29 +34,32 @@ public function getName()
 
     /**
      * @param DistinguishedName $distinguishedName
+     * @param bool $disableAlternateDomains
      *
      * @return bool
      */
-    public function isReadyForRequest(DistinguishedName $distinguishedName)
+    public function isReadyForRequest(DistinguishedName $distinguishedName, $disableAlternateDomains)
     {
         return $distinguishedName->getCountryName()
             && $distinguishedName->getStateOrProvinceName()
             && $distinguishedName->getLocalityName()
             && $distinguishedName->getOrganizationName()
             && $distinguishedName->getOrganizationalUnitName()
             && $distinguishedName->getEmailAddress()
-            && $distinguishedName->getCommonName();
+            && $distinguishedName->getCommonName()
+            && $disableAlternateDomains;
```

Comment:
That's mean, certificate with alternateSubjectName will never be ready?
What's about ignoring SANs and automatically upgrading the DistinguishedName ? (see my PR)